util/util.go

```diff
@@ -221,15 +234,47 @@ func MakeDeploy(
 
 	deploy = &ipc.DeployItem{
 		Address:           fromAddress,
-		Session:           &ipc.DeployPayload{Payload: &ipc.DeployPayload_DeployCode{DeployCode: &ipc.DeployCode{Code: sessionCode, Args: sessionArgs}}},
-		Payment:           &ipc.DeployPayload{Payload: &ipc.DeployPayload_DeployCode{DeployCode: &ipc.DeployCode{Code: paymentCode, Args: paymentArgs}}},
+		Session:           MakeDeployPayload(sessionName, sessionData, sessionArgs),
+		Payment:           MakeDeployPayload(paymentName, paymentData, paymentArgs),
 		GasPrice:          gasPrice,
 		AuthorizationKeys: [][]byte{fromAddress},
 		DeployHash:        headerHash}
 
 	return deploy
 }
 
+func MakeDeployCode(name string, data []byte, args []byte) *consensus.Deploy_Code {
```

Comment:
How about using iota instead of string?

Reply:
It is good to use iota. 

Regarding data, name is a string and another everything is []byte. Is there a good way to use this? (ie. (type iota, stringData string, bytesData []byte, args []byte))

Reply:
I think like below
```go
func MakeDeployPayload(kind DeployType, data []byte, args []byte) *ipc.DeployPayload {
	deployPayload := &ipc.DeployPayload{}
	switch kind {
	case Wasm:
		deployPayload.Payload = &ipc.DeployPayload_DeployCode{DeployCode: &ipc.DeployCode{Code: data, Args: args}}
	case Uref:
		deployPayload.Payload = &ipc.DeployPayload_StoredContractUref{StoredContractUref: &ipc.StoredContractURef{Uref: data, Args: args}}
	case Hash:
		deployPayload.Payload = &ipc.DeployPayload_StoredContractHash{StoredContractHash: &ipc.StoredContractHash{Hash: data, Args: args}}
        case Name:
                deployPayload.Payload = &ipc.DeployPayload_StoredContractName{StoredContractName: &ipc.StoredContractName{StoredContractName: data, Args: args}}
	default:
		panic("Unknow Deploy Type"); //or propagate error
	}

	return deployPayload
}
```

Reply:
StoredContractName type is not []byte. that is string.

Reply:
You can simply convert the data into string.

Reply:
Probably user or something will be entered as a string. Do you want convert it to byte and then convert it back to string when you use it again? If you think that's no problem, I'll do apply this one.

Reply:
Yap~! I think that's the way to go, because the purpose of the first arg is not to provide data but to discern payload type. And also the invalid input should be handled in default case.

But if you still wonder whether it is right decision, it would be good to request another review :).

Reply:
https://github.com/hdac-io/casperlabs-ee-grpc-go-util/pull/7/commits/900e086df8a501abf94e88e99bd102297c4a4b94

It's nice to use iota, but I don't think it's bad to use the string form used by cli.

Reply:
I'm also standing in iota.
I also think switching flag should work only for flag, not act for other params.

But it seems to be frustrated to increase arguments or conversion.
Then, how about using struct?

```golang
type DeployCodeInput struct {
    Bytes []byte
    Strings string
}

func MakeDeployCode(kindof string, data DeployInput, args []byte) *consensus.Deploy_Code {
	deployCode := &consensus.Deploy_Code{AbiArgs: args}
	switch kindof {
	case "wasm":
		deployCode.Contract = &consensus.Deploy_Code_Wasm{Wasm: DeployInput.Bytes}
	case "uref":
		deployCode.Contract = &consensus.Deploy_Code_Uref{Uref: DeployInput.Bytes}
	case "hash":
		deployCode.Contract = &consensus.Deploy_Code_Hash{Hash: DeployInput.Bytes}
	case "name":
		deployCode.Contract = &consensus.Deploy_Code_Name{Name: DeployInput.Strings}
        default:
                // cover error case
        }

	return deployCode
}
```

Reply:
Or struct wrapper

```golang
type DeployCodeInput struct {
    consensus.Deploy_Code_Wasm
}
```

Reply:
https://github.com/hdac-io/casperlabs-ee-grpc-go-util/pull/7/commits/840d4cf75651a0ce2874f5afa761502538b00f29

@psy2848048 I think using cpu(converting) is better  than memory(add struct). What do you think? @hdac-io/developers

Reply:
casting between []bytes and string in golang is tiny as far as I know.
LGTM!

Reply:
No objection more :)